components/OtherResourcesForm/OtherResourcesForm.tsx

```diff
@@ -1,27 +1,148 @@
-import clsx from "clsx";
-import { useTranslation } from "react-i18next";
-import Textarea from "../Form/Textarea";
+import clsx from 'clsx'
+import { useTranslation } from 'react-i18next'
+import Textarea from '../Form/Textarea'
+import { useOthersForm } from '@/hooks/useData'
+import { useForm } from 'react-hook-form'
+import DateInput from '@/components/Form/Date'
+import Dropdown from '@/components/Form/Dropdown'
+import Input from '@/components/Form/Input'
+import endpoints from '../../endpoints.json'
+import { useState } from 'react'
 
 const OtherResourcesForm = ({}) => {
-  const { t } = useTranslation();
+  const { t } = useTranslation()
+  const { data: formData } = useOthersForm()
+  const [serverErrors, setServerErrors] = useState<{ [key: string]: string[] }>(
+    {}
+  )
+  const countyCovarage = formData ? formData['county_coverage'].choices : []
+  const today = new Date().toISOString().substr(0, 10)
+  const {
+    handleSubmit,
+    register,
+    formState: { errors },
+  } = useForm()
```

Comment:
seems like you have predefined input fields, you could define an interface or a type with all the fields and use it asa  generic here. Also below on submit, you can use that interface you defined above instead of `any`